c++/src/kj/compat/http.c++

```diff
@@ -3135,10 +3135,13 @@ private:
           compressor.reset();
         }
         auto& innerMessage = compressedMessage.emplace(compressor.processMessage(message));
-        KJ_ASSERT(innerMessage.asPtr().endsWith({0x00, 0x00, 0xFF, 0xFF}));
-        message = innerMessage.slice(0, innerMessage.size() - 4);
-        // Strip 0x00 0x00 0xFF 0xFF off the tail.
-        // See: https://datatracker.ietf.org/doc/html/rfc7692#section-7.2.1
+        message = innerMessage;
+        if (innerMessage.size() > 0) {
+          KJ_ASSERT(innerMessage.asPtr().endsWith({0x00, 0x00, 0xFF, 0xFF}));
+          message = innerMessage.slice(0, innerMessage.size() - 4);
+          // Strip 0x00 0x00 0xFF 0xFF off the tail.
+          // See: https://datatracker.ietf.org/doc/html/rfc7692#section-7.2.1
+        }
```

Comment:
With the context of the previous comment, we could change this to:

```
if (message.size() > 0) {
  KJ_ASSERT(innerMessage.asPtr().endsWith({0x00, 0x00, 0xFF, 0xFF}));
  message = innerMessage.slice(0, innerMessage.size() - 4);
  // Strip 0x00 0x00 0xFF 0xFF off the tail.
  // See: https://datatracker.ietf.org/doc/html/rfc7692#section-7.2.1
} else {
  // We didn't compress anything so we add an empty deflate block.
  message = compressedMessage.emplace(Array<kj::byte>(0x00));
}
```

The (new) test still passes, and it seems to be closer to what the spec asks for?